OracleGoldenGate/23/bin/deployment-main.sh

```diff
@@ -0,0 +1,204 @@
+
+    JAVA_HOME="$(dirname "$(dirname "$(readlink -f "${java}")")")"
+    export JAVA_HOME
+}
+
+##
+##  l o c a t e _ l  i  b _  j  v  m
+##  Locate the shared library libjvm.so and set LD_LIBRARY_PATH
+##
+function locate_lib_jvm() {
+    [[ -z "${JAVA_HOME}" ]] && abort "Java installation not found"
+
+    local libjvm
+    libjvm="$(find "${JAVA_HOME}" -name libjvm.so | head -1)"
+    if [ -z "${libjvm}" ]; then
+        echo "Warning: The shared library libjvm.so cannot be located."
+    else
+        local JVM_LIBRARY_PATH
+        JVM_LIBRARY_PATH="$(dirname "${libjvm}" )"
+        export LD_LIBRARY_PATH=$JVM_LIBRARY_PATH:$LD_LIBRARY_PATH
+    fi
+}
+
+##
+##  r u n _ a s _ o g g
+##  Return a string used for running a process as the 'ogg' user
+##
+function run_as_ogg() {
+    local user="ogg"
+    local uid gid
+    uid="$(id -u "${user}")"
+    gid="$(id -g "${user}")"
+    echo "setpriv --ruid ${uid} --euid ${uid} --groups ${gid} --rgid ${gid} --egid ${gid} -- "
+}
+
+##
+##  s e t u p _ d e p l o y m e n t _ d i r e c t o r i e s
+##  Create and set permissions for directories for the deployment
+##
+function setup_deployment_directories() {
+    rm    -fr        "${OGG_DEPLOYMENT_HOME}"/Deployment/var/{run,temp,lib/db} \
+                     "${OGG_TEMPORARY_FILES}"/{run,temp}
+    mkdir -p         "${OGG_TEMPORARY_FILES}"/{run,temp,db} \
+                     "${OGG_DEPLOYMENT_HOME}"/Deployment/var/lib
+    ln -s            "${OGG_TEMPORARY_FILES}"/run  "${OGG_DEPLOYMENT_HOME}"/Deployment/var/run
+    ln -s            "${OGG_TEMPORARY_FILES}"/temp "${OGG_DEPLOYMENT_HOME}"/Deployment/var/temp
+    ln -s            "${OGG_TEMPORARY_FILES}"/db   "${OGG_DEPLOYMENT_HOME}"/Deployment/var/lib/db
+
+    chown    ogg:ogg "${OGG_DEPLOYMENT_HOME}" "${OGG_TEMPORARY_FILES}"
+    chmod    0750    "${OGG_DEPLOYMENT_HOME}" "${OGG_TEMPORARY_FILES}"
+    find             "${OGG_DEPLOYMENT_HOME}" "${OGG_TEMPORARY_FILES}" -mindepth 1 -maxdepth 1 -not -name '.*' -exec \
+    chown -R ogg:ogg {} \;
+}
+
+##
+##  r u n _ u s e r _ s c r i p t s
+##  Run custom scripts in the container before and after GoldenGate starts
+##
+function run_user_scripts {
+    local scripts="${1}";
+    while read -r script; do
+        case "${script}" in
+            *.sh)
+                echo "Running script '${script}'"
+                # shellcheck disable=SC1090
+                source "${script}"
+                ;;
+            *)
+                echo "Ignoring '${script}'"
+                ;;
+        esac
+    done < <(find "${scripts}" -type f | sort)
+}
+
+##
+##  s t a r t _ o g g
+##  Initialize and start the OGG installation
+##
+function start_ogg() {
+    $(run_as_ogg) python3 /usr/local/bin/deployment-init.py
+    $(run_as_ogg) tail -F "${OGG_DEPLOYMENT_HOME}"/ServiceManager/var/log/ServiceManager.log &
+    ogg_pid=$!
+}
+
+##
+##  s t a r t _ n g i n x
+##  Start the NGinx reverse proxy daemon
```

Comment:
The correct way to format NGINX is in all capitals as per their trademark documentation: https://www.f5.com/company/policies/trademarks